src/newgrf.cpp

```diff
@@ -9161,6 +9161,13 @@ static void CalculateRefitMasks()
 				ei->cargo_type = (CargoID)FindFirstBit(ei->refit_mask);
 			}
 		}
+		if (!IsValidCargoID(ei->cargo_type) && e->type == VEH_TRAIN && e->u.rail.railveh_type != RAILVEH_WAGON && e->u.rail.capacity == 0) {
+			/* For train engines which do not carry cargo it does not matter if their cargo type is invalid, fallback to
+			 * the first available instead, if the cargo type has not been changed. */
```

Comment:
Ghehe, this is a very long sentence :)

```suggestion
			/* For train engines which do not carry cargo it does not matter if their cargo type is invalid.
			 * Fallback to the first available instead, if the cargo type has not been changed. */
```